zer0-os/zOS
# feat: app switcher

### What does this do?

- Creates a top-level `App` component, which houses all "global UI" (providers, app bar), and renders the `AppSwitcher`.
- Creates an `AppRouter`, which is basically just a `Router` which renders different apps to a container based on `Route`.
- Moves all "global UI" from `Main` to `App`.
- Adds a feature flag (`enableExplorer`) for testing the app switcher.

### Why are we making this change?

So we can add arbitrary "sub apps" to zOS, as per the initial vision of the product.

### How do I test this?

- These changes should deliver 0 perceivable difference to the user. So, all flows should work exactly the same as before.
- Run `FEATURE_FLAGS.enableExplorer = true` in your browser console, and click the `Explorer` option in the app bar. Note: you will have to hit `Back` in your browser to go back to Messenger

---

Please note:
- I used function components because both of these components are very small and don't contain much business logic. I'm happy to rewrite them as class components if need be, 
- I used `@testing-library/react` for unit tests, as we are meant to be moving away from Enzyme.
